src/main/java/com/iota/iri/Milestone.java

```diff
@@ -115,6 +137,12 @@ public void init(final SpongeFactory.Mode mode, final LedgerValidator ledgerVali
                                                 }
                                                 break;
                                             case INCOMPLETE:
+                                                // issue a solidity check to solidify incomplete milestones
+                                                // Note: otherwise a milestone that was followed by a coo-snapshot might
+                                                //       never get solidifed again since it doesnt have connections to
+                                                //       the tips
+                                                transactionValidator.checkSolidity(t.getHash(), true);
+
```

Comment:
IRI doesn't persist the tips between iri restarts. So if you restart IRI or happen to download a database with an incomplete milestone it will never be "fixed".

Reply:
I see you added it to the `unsolidMilestones` set which is way better.
I guess this change is necessary until we sort out solidifications...
@karimodm note my previous comment